format-common/src/main/java/io/cdap/plugin/format/input/PathTrackingInputFormatProvider.java

```diff
@@ -63,6 +65,31 @@ protected void validate() {
     // no-op
   }
 
+  protected Schema addPathField(Schema dataSchema, FailureCollector failureCollector) {
+    return addPathField(failureCollector, dataSchema, conf.getPathField());
+  }
+
+  public static Schema addPathField(FailureCollector failureCollector, Schema dataSchema, String pathField) {
+    if (pathField == null) {
+      return dataSchema;
+    }
+    if (dataSchema == null) {
+      failureCollector.addFailure("Schema is null", "Please provide valid schema")
+        .withConfigProperty("nullSchema");
+      throw failureCollector.getOrThrowException();
+    }
+    List<Schema.Field> fields = new ArrayList<>(dataSchema.getFields());
+    for (Schema.Field field : fields) {
+      if (pathField.equals(field.getName())) {
+        failureCollector.addFailure("Pathfield already exists in the schema",
```

Comment:
nit: Path Field %s already exists in the data schema.

Reply:
Done